App Submission: Bitcoin Regtest Dashboard

coreyphillips opened the pull request:
# App Submission

### App name
Bitcoin Regtest Dashboard

## Icon
![icon](https://raw.githubusercontent.com/coreyphillips/bitcoin-regtest-dashboard/main/icon.svg)

## Gallery Screenshots
![screenshot1](https://raw.githubusercontent.com/coreyphillips/bitcoin-regtest-dashboard/main/gallery/screenshot1.png)
![screenshot2](https://raw.githubusercontent.com/coreyphillips/bitcoin-regtest-dashboard/main/gallery/screenshot2.png)
![screenshot3](https://raw.githubusercontent.com/coreyphillips/bitcoin-regtest-dashboard/main/gallery/screenshot3.png)
![screenshot4](https://raw.githubusercontent.com/coreyphillips/bitcoin-regtest-dashboard/main/gallery/screenshot4.png)

### I have tested my app on:
- [x] umbrelOS on a Raspberry Pi
- [x] umbrelOS on an Umbrel Home
- [x] umbrelOS on Linux VM

### Description
A self-contained development dashboard with its own Bitcoin Core regtest node and Electrs instance. Perfect for Bitcoin developers who need quick access to regtest functionality for testing wallets and applications.

### Features
- Mine blocks instantly
- Generate addresses
- Send Regtest Bitcoin
- RBF transactions
- View transactions
- Observe Mempool & Blocks
- Search Addresses & Transactions via the built-in explorer
- Connect external wallets via built in Electrs
- Decode & Broadcast raw transactions
- Full RPC console

coreyphillips committed:
App Submission: Bitcoin Regtest Dashboard

coreyphillips committed:
Update submission URL

coreyphillips committed:
fix: linting errors

- Fix linting errors: category, port, gallery, data dirs

al-lac reviewed (changes requested):
Hey @coreyphillips,

thanks a lot for submitting your app to the Umbrel App Store, we really appreciate it! 🤝 

Really cool app for testing out various scenarios and a nice looking dashboard 👍 

Added some comments on what should be adjusted to get the app ready for primetime.

coreyphillips committed:
Update bitcoin-regtest-dashboard/umbrel-app.yml

Co-authored-by: Alexander L. <lackner.alex@gmail.com>

coreyphillips committed:
fix: address PR review feedback

- Remove icon.svg
- Update to v1.0.1
- Expose Electrs on port 60401

coreyphillips committed:
fix: address PR review feedback

- Remove unused Bitcoin expose ports

coreyphillips commented on bitcoin-regtest-dashboard/icon.svg:
Removed icon as of [this](https://github.com/getumbrel/umbrel-apps/pull/4264/commits/ea748e826d8f748882a47dc3fb3bd4846f525fff) commit.

coreyphillips committed:
fix: address PR review feedback

- Add icon and gallery URLs
- Update dashboard image to v1.0.2

coreyphillips commented:
> Hey @coreyphillips, thanks a lot for making all of the adaptions, great job!
> 
> Only added one more comment about the used bitcoin version.
> 
> Can you also please attach an icon and some more screenshots?

@al-lac Of course, I added the icon and screenshots [here](https://github.com/getumbrel/umbrel-apps/pull/4264/commits/31bff7aa24a0c9bce153589ebc8e4927c3b2526e).

al-lac commented on bitcoin-regtest-dashboard/docker-compose.yml:
```diff
@@ -0,0 +1,89 @@
+
+  # Bitcoin Core in regtest mode (self-contained, no dependencies)
+  bitcoin:
+    image: getumbrel/bitcoind:v28.1@sha256:c565266ea302c9ab2fc490f04ff14e584210cde3d0d991b8309157e5dfae9e8d
```
Great! Can you also update it here on the submission?

coreyphillips replied:
Yes, updated [here](https://github.com/getumbrel/umbrel-apps/pull/4264/commits/f6aa74a48a644bd378de0c4563090955435e2908) as well. Thanks!

coreyphillips committed:
fix: address PR review feedback

- Remove icon and gallery URLs

coreyphillips committed:
fix: address PR review feedback

- Switch from getumbrel/bitcoind:v28.1 to ghcr.io/getumbrel/docker-bitcoind:v30.0

al-lac committed:
Add gallery assets

al-lac commented:
Hey @coreyphillips, thanks a lot again for providing bitcoin-regtest-dashboard to the Umbrel App Store! 🔝 

I just added links to to the gallery assets, tested the app one more time and we are ready to go live 🎉
